toolchain/base/value_store.h

```diff
@@ -33,20 +33,16 @@ class ValueStoreNotPrintable {};
 
 // A simple wrapper for accumulating values, providing IDs to later retrieve the
 // value. This does not do deduplication.
-//
-// IdT::ValueType must represent the type being indexed.
-template <typename IdT>
-  requires(requires { typename IdT::ValueType; })
+template <typename IdT, typename ValueT>
```

Comment:
As a transition measure, perhaps this could have a default value? (`template <typename IdT, typename ValueT = IdT::ValueType>`)

Reply:
All uses are fixed by this PR; I'm using it to find uses and to prevent regressions. If you're alright with it, I'd rather not add a default.